apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -112,6 +132,28 @@ export class MessageMatcher {
     return !!(await findAsync(asyncFilters, (i) => this.processFilter(variables, i, command)));
   }
 
+  private async processIsOnline(
+    filter: IRealtimeOnlineFilterPart | IOnlineInLastFilterPart,
+    command: SendMessageCommand
+  ): Promise<boolean> {
+    const subscriber = await this.subscriberRepository.findOne({
+      _id: command.subscriberId,
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+    });
+
+    const isOnlineMatch = subscriber?.isOnline === filter.value;
+    if (filter.on === 'isOnline') {
+      return isOnlineMatch;
+    }
```

Comment:
for the `isOnline` filter we just do check if the `isOnline` value on the subscriber matches with the filter value